gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -242,28 +242,45 @@ def self._on_method_added(hook_mod, method_name, is_singleton_method: false)
     end
   end
 
-  def self._handle_missing_method_signature(klass, original_method, callee)
+  def self._handle_missing_method_signature(receiver, original_method, callee)
     method_sig = T::Private::Methods.signature_for_method(original_method)
+    if !method_sig
+      raise "`sig` not present for method `#{callee}` on #{receiver.inspect} but you're trying to run it anyways. " \
+        "This should only be executed if you used `alias_method` to grab a handle to a method after `sig`ing it, but that clearly isn't what you are doing. " \
+        "Maybe look to see if an exception was thrown in your `sig` lambda or somehow else your `sig` wasn't actually applied to the method."
+    end
 
-    aliasing_method = klass.instance_method(callee)
-    aliasing_mod = aliasing_method.owner
+    if receiver.class <= original_method.owner
```

Comment:
Doing this instead of `is_singleton_method ? self.singleton_class : self.class` - and just ignoring `is_singleton_method` - is the fix for **3** from the PR description.